src/persist/src/indexed/trace.rs

```diff
@@ -278,23 +278,12 @@ impl BlobTrace {
     }
 
     /// Update the compaction frontier to `since`.
-    pub fn allow_compaction(&mut self, since: Antichain<u64>) -> Result<(), Error> {
-        if PartialOrder::less_equal(&self.seal, &since) {
-            return Err(Error::from(format!(
-                "invalid compaction at or in advance of trace seal {:?}: {:?}",
-                self.seal, since,
-            )));
-        }
-
-        if PartialOrder::less_equal(&since, &self.since) {
-            return Err(Error::from(format!(
-                "invalid compaction less than or equal to trace since {:?}: {:?}",
-                self.since, since
-            )));
-        }
-
-        self.since = since;
-        Ok(())
+    ///
+    /// This function intentionally does not do any checking to see if ts is
```

Comment:
Hmm, I don't love this, even as a stopgap. I worry that we'll forget to circle back and clean it up when we do the real fix. How hard would it be to grab the serialization of the state in trace at the beginning of Indexed::allow_compaction and use that to wholesale replace the trace in the traces map on Indexed? Sort of like a mini version of what we're about to do in the batch PR

Reply:
done